src/extension.ts

```diff
@@ -198,32 +205,43 @@ async function selectMplabxProjectFolder(): Promise<string | undefined> {
 }
 
 export function deactivate() {
+	killAllMdbProcess();
+}
+
+function killAllMdbProcess() {
+	mdbDebugSession.forEach((v, k) => v.shutdown(true));
+	mdbDebugSession.clear();
 }
 
 class MdbDebugAdapterServerDescriptorFactory implements vscode.DebugAdapterDescriptorFactory {
 
 	private server?: Net.Server;
 
 	createDebugAdapterDescriptor(session: vscode.DebugSession, executable: vscode.DebugAdapterExecutable | undefined): vscode.ProviderResult<vscode.DebugAdapterDescriptor> {
-
+		// TODO: Figure out how to close debugSession properly
 		if (!this.server) {
 			// start listening on a random port
 			this.server = Net.createServer(socket => {
-				const session = new MdbDebugSession();
-				session.setRunAsServer(true);
-				session.start(socket as NodeJS.ReadableStream, socket);
+				let debugSession: MdbDebugSession;
+				const configArgs = session.configuration as MplabxDebugConfiguration;
+				if (configArgs.type === 'mdb' && configArgs.runMdbAsServer) {
```

Comment:
Is there a reason `runAsServer` only works with type 'mdb'?

Reply:
Because we haven't used the programmer yet and are currently using the simulator and assembly language, I can't test anything else yet.
And also because I hope it won't affect the original functionality, so I added this setting.
In addition, I don’t know why it takes a long time to open mdb, so if you use `runMdbAsServer`, it will open a resident process until the project is closed, when there is a new debug session, the mdb process will be attach to new session.

Reply:
That all makes sense. I have noticed that the MDB takes a long time to get started was well. Running as a server is a good solution to the problem. That being said, you didn't answer the question of why the current implication only works when the `mdb` profile is used directly. It should also work if a `mplabx` profile is used. Is there a reason this wouldn't work with the `mplabx` debug type?